<div style="border:solid steelblue 3px; padding: 20px; border-radius: 10px">

**Привет, Матвей!** 👋

Меня зовут Алексей Манелюк, я буду проверять твой проект. В Яндекс Практикуме принято общаться на "ты" 😉, но если некомфортно - скажи и перейдем на "Вы".

Моя цель - не указать на ошибки, а поделиться опытом и помочь в освоении анализа данных.

В тетрадке ты увидишь мои комментарии, пожалуйста, не удаляй, не перемещай и не изменяй их. У комментариев могут быть приписки **V2**, **V3**, **V...**, которые соответствуют каждой версии проверки. 
    
Я буду использовать цветовую разметку:
<p>
<div class="alert-danger" style="padding: 20px; border-radius: 10px">
<b>Комментарий ревьюера ❌</b>

Так выделены критические замечания. Без их отработки я не смогу принять проект. </div>

<div class="alert-warning" style="padding: 20px; border-radius: 10px">
<b>  Комментарий ревьюера ⚠️</b>

Так выделены небольшие замечания или полезные советы. Постарайся учесть их.</div>
<p>
<div class="alert-success" style="padding: 20px; border-radius: 10px">
<b>Комментарий ревьюера ✔️</b>

Так выделены правильные решения.</div>

Предлагаю работать в диалоге: если что-то дорабатываешь или отвечаешь на мой комментарий - пиши об этом. Мне будет удобней, если ты выделишь свои комментарии:
<p>
<div class="alert-info" style="padding: 20px; border-radius: 10px">
<b>Комментарий студента 💎</b>

Здесь исправил.</div>

Все это поможет мне быстрее принять твой проект.</div>

<div class="alert-info" style="padding: 20px; border-radius: 10px">
<b>Комментарий студента 💎</b>

Написать свой комментарий можно, скопировав этот блок в ячейку Markdown.</div>

# Исследование надежности заемщиков


Во второй части проекта вы выполните шаги 3 и 4. Их вручную проверит ревьюер.
Чтобы вам не пришлось писать код заново для шагов 1 и 2, мы добавили авторские решения в ячейки с кодом. 



<div class="alert-warning" style="padding: 20px; border-radius: 10px">
<b>Комментарий ревьюера ⚠️</b>

Несмотря на то, что первая часть проекта проверяется автоматизрованно, дам по ней пару советов - они пригодятся тебе в следующих проектах. 

**Их отработка сейчас необязательна**.

📌 После названия, обязательно нужно добавлять бриф, цели исследования, описание данных и план работы.</div>

## Откройте таблицу и изучите общую информацию о данных

**Задание 1. Импортируйте библиотеку pandas. Считайте данные из csv-файла в датафрейм и сохраните в переменную `data`. Путь к файлу:**

`/datasets/data.csv`

In [1]:
import pandas as pd

try:
    data = pd.read_csv('/datasets/data.csv')
except:
    data = pd.read_csv('https://code.s3.yandex.net/datasets/data.csv')

**Задание 2. Выведите первые 20 строчек датафрейма `data` на экран.**

In [2]:
data.head(20)

children  days_employed  dob_years            education  education_id  \
0          1   -8437.673028         42               высшее             0   
1          1   -4024.803754         36              среднее             1   
2          0   -5623.422610         33              Среднее             1   
3          3   -4124.747207         32              среднее             1   
4          0  340266.072047         53              среднее             1   
5          0    -926.185831         27               высшее             0   
6          0   -2879.202052         43               высшее             0   
7          0    -152.779569         50              СРЕДНЕЕ             1   
8          2   -6929.865299         35               ВЫСШЕЕ             0   
9          0   -2188.756445         41              среднее             1   
10         2   -4171.483647         36               высшее             0   
11         0    -792.701887         40              среднее             1   
12         0            NaN         65              среднее             1   
13         0   -1846.641941         54  неоконченное высшее             2   
14         0   -1844.956182         56               высшее             0   
15         1    -972.364419         26              среднее             1   
16         0   -1719.934226         35              среднее             1   
17         0   -2369.999720         33               высшее             0   
18         0  400281.136913         53              среднее             1   
19         0  -10038.818549         48              СРЕДНЕЕ             1   

       family_status  family_status_id gender income_type  debt  \
0    женат / замужем                 0      F   сотрудник     0   
1    женат / замужем                 0      F   сотрудник     0   
2    женат / замужем                 0      M   сотрудник     0   
3    женат / замужем                 0      M   сотрудник     0   
4   гражданский брак                 1      F   пенсионер     0   
5   гражданский брак                 1      M   компаньон     0   
6    женат / замужем                 0      F   компаньон     0   
7    женат / замужем                 0      M   сотрудник     0   
8   гражданский брак                 1      F   сотрудник     0   
9    женат / замужем                 0      M   сотрудник     0   
10   женат / замужем                 0      M   компаньон     0   
11   женат / замужем                 0      F   сотрудник     0   
12  гражданский брак                 1      M   пенсионер     0   
13   женат / замужем                 0      F   сотрудник     0   
14  гражданский брак                 1      F   компаньон     1   
15   женат / замужем                 0      F   сотрудник     0   
16   женат / замужем                 0      F   сотрудник     0   
17  гражданский брак                 1      M   сотрудник     0   
18    вдовец / вдова                 2      F   пенсионер     0   
19         в разводе                 3      F   сотрудник     0   

     total_income                                 purpose  
0   253875.639453                           покупка жилья  
1   112080.014102                 приобретение автомобиля  
2   145885.952297                           покупка жилья  
3   267628.550329              дополнительное образование  
4   158616.077870                         сыграть свадьбу  
5   255763.565419                           покупка жилья  
6   240525.971920                       операции с жильем  
7   135823.934197                             образование  
8    95856.832424                   на проведение свадьбы  
9   144425.938277                 покупка жилья для семьи  
10  113943.491460                    покупка недвижимости  
11   77069.234271       покупка коммерческой недвижимости  
12            NaN                         сыграть свадьбу  
13  130458.228857                 приобретение автомобиля  
14  165127.911772              покупка жилой недвижимости  
15  116820.904450 

**Задание 3. Выведите основную информацию о датафрейме с помощью метода `info()`.**

In [3]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21525 entries, 0 to 21524
Data columns (total 12 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   children          21525 non-null  int64  
 1   days_employed     19351 non-null  float64
 2   dob_years         21525 non-null  int64  
 3   education         21525 non-null  object 
 4   education_id      21525 non-null  int64  
 5   family_status     21525 non-null  object 
 6   family_status_id  21525 non-null  int64  
 7   gender            21525 non-null  object 
 8   income_type       21525 non-null  object 
 9   debt              21525 non-null  int64  
 10  total_income      19351 non-null  float64
 11  purpose           21525 non-null  object 
dtypes: float64(2), int64(5), object(5)
memory usage: 2.0+ MB


<div class="alert-warning" style="padding: 20px; border-radius: 10px">
<b>Комментарий ревьюера ⚠️</b>

📌 Каждый раздел работы стоит завершать кратким выводом, о том что было сделано и какие результаты получены.

Как правильно оформить проект ты можешь прочитать в материалах курса:

`Полезные инструкции для учёбы - Оформление проекта - Рекомендации по выполнению проектов`
</div>

## Предобработка данных

### Удаление пропусков

**Задание 4. Выведите количество пропущенных значений для каждого столбца. Используйте комбинацию двух методов.**

In [4]:
data.isna().sum()

children               0
days_employed       2174
dob_years              0
education              0
education_id           0
family_status          0
family_status_id       0
gender                 0
income_type            0
debt                   0
total_income        2174
purpose                0
dtype: int64

**Задание 5. В двух столбцах есть пропущенные значения. Один из них — `days_employed`. Пропуски в этом столбце вы обработаете на следующем этапе. Другой столбец с пропущенными значениями — `total_income` — хранит данные о доходах. На сумму дохода сильнее всего влияет тип занятости, поэтому заполнить пропуски в этом столбце нужно медианным значением по каждому типу из столбца `income_type`. Например, у человека с типом занятости `сотрудник` пропуск в столбце `total_income` должен быть заполнен медианным доходом среди всех записей с тем же типом.**

In [5]:
for t in data['income_type'].unique():
    data.loc[(data['income_type'] == t) & (data['total_income'].isna()), 'total_income'] = \
    data.loc[(data['income_type'] == t), 'total_income'].median()

### Обработка аномальных значений

**Задание 6. В данных могут встречаться артефакты (аномалии) — значения, которые не отражают действительность и появились по какой-то ошибке. таким артефактом будет отрицательное количество дней трудового стажа в столбце `days_employed`. Для реальных данных это нормально. Обработайте значения в этом столбце: замените все отрицательные значения положительными с помощью метода `abs()`.**

In [6]:
data['days_employed'] = data['days_employed'].abs()

**Задание 7. Для каждого типа занятости выведите медианное значение трудового стажа `days_employed` в днях.**

In [7]:
data.groupby('income_type')['days_employed'].agg('median')

income_type
безработный        366413.652744
в декрете            3296.759962
госслужащий          2689.368353
компаньон            1547.382223
пенсионер          365213.306266
предприниматель       520.848083
сотрудник            1574.202821
студент               578.751554
Name: days_employed, dtype: float64

У двух типов (безработные и пенсионеры) получатся аномально большие значения. Исправить такие значения сложно, поэтому оставьте их как есть. Тем более этот столбец не понадобится вам для исследования.

**Задание 8. Выведите перечень уникальных значений столбца `children`.**

In [8]:
data['children'].unique()

array([ 1,  0,  3,  2, -1,  4, 20,  5])

**Задание 9. В столбце `children` есть два аномальных значения. Удалите строки, в которых встречаются такие аномальные значения из датафрейма `data`.**

In [9]:
data = data[(data['children'] != -1) & (data['children'] != 20)]

**Задание 10. Ещё раз выведите перечень уникальных значений столбца `children`, чтобы убедиться, что артефакты удалены.**

In [10]:
data['children'].unique()

array([1, 0, 3, 2, 4, 5])

### Удаление пропусков (продолжение)

**Задание 11. Заполните пропуски в столбце `days_employed` медианными значениями по каждого типа занятости `income_type`.**

In [11]:
for t in data['income_type'].unique():
    data.loc[(data['income_type'] == t) & (data['days_employed'].isna()), 'days_employed'] = \
    data.loc[(data['income_type'] == t), 'days_employed'].median()

**Задание 12. Убедитесь, что все пропуски заполнены. Проверьте себя и ещё раз выведите количество пропущенных значений для каждого столбца с помощью двух методов.**

In [12]:
data.isna().sum()

children            0
days_employed       0
dob_years           0
education           0
education_id        0
family_status       0
family_status_id    0
gender              0
income_type         0
debt                0
total_income        0
purpose             0
dtype: int64

### Изменение типов данных

**Задание 13. Замените вещественный тип данных в столбце `total_income` на целочисленный с помощью метода `astype()`.**

In [13]:
data['total_income'] = data['total_income'].astype(int)

### Обработка дубликатов

**Задание 14. Обработайте неявные дубликаты в столбце `education`. В этом столбце есть одни и те же значения, но записанные по-разному: с использованием заглавных и строчных букв. Приведите их к нижнему регистру. Проверьте остальные столбцы.**

In [14]:
data['education'] = data['education'].str.lower()

**Задание 15. Выведите на экран количество строк-дубликатов в данных. Если такие строки присутствуют, удалите их.**

In [15]:
data.duplicated().sum()

71

In [16]:
data = data.drop_duplicates()

### Категоризация данных

**Задание 16. На основании диапазонов, указанных ниже, создайте в датафрейме `data` столбец `total_income_category` с категориями:**

- 0–30000 — `'E'`;
- 30001–50000 — `'D'`;
- 50001–200000 — `'C'`;
- 200001–1000000 — `'B'`;
- 1000001 и выше — `'A'`.


**Например, кредитополучателю с доходом 25000 нужно назначить категорию `'E'`, а клиенту, получающему 235000, — `'B'`. Используйте собственную функцию с именем `categorize_income()` и метод `apply()`.**

In [17]:
def categorize_income(income):
    try:
        if 0 <= income <= 30000:
            return 'E'
        elif 30001 <= income <= 50000:
            return 'D'
        elif 50001 <= income <= 200000:
            return 'C'
        elif 200001 <= income <= 1000000:
            return 'B'
        elif income >= 1000001:
            return 'A'
    except:
        pass

In [18]:
data['total_income_category'] = data['total_income'].apply(categorize_income)

**Задание 17. Выведите на экран перечень уникальных целей взятия кредита из столбца `purpose`.**

In [19]:
data['purpose'].unique()

array(['покупка жилья', 'приобретение автомобиля',
       'дополнительное образование', 'сыграть свадьбу',
       'операции с жильем', 'образование', 'на проведение свадьбы',
       'покупка жилья для семьи', 'покупка недвижимости',
       'покупка коммерческой недвижимости', 'покупка жилой недвижимости',
       'строительство собственной недвижимости', 'недвижимость',
       'строительство недвижимости', 'на покупку подержанного автомобиля',
       'на покупку своего автомобиля',
       'операции с коммерческой недвижимостью',
       'строительство жилой недвижимости', 'жилье',
       'операции со своей недвижимостью', 'автомобили',
       'заняться образованием', 'сделка с подержанным автомобилем',
       'получение образования', 'автомобиль', 'свадьба',
       'получение дополнительного образования', 'покупка своего жилья',
       'операции с недвижимостью', 'получение высшего образования',
       'свой автомобиль', 'сделка с автомобилем',
       'профильное образование', 'высшее об

**Задание 18. Создайте функцию, которая на основании данных из столбца `purpose` сформирует новый столбец `purpose_category`, в который войдут следующие категории:**

- `'операции с автомобилем'`,
- `'операции с недвижимостью'`,
- `'проведение свадьбы'`,
- `'получение образования'`.

**Например, если в столбце `purpose` находится подстрока `'на покупку автомобиля'`, то в столбце `purpose_category` должна появиться строка `'операции с автомобилем'`.**

**Используйте собственную функцию с именем `categorize_purpose()` и метод `apply()`. Изучите данные в столбце `purpose` и определите, какие подстроки помогут вам правильно определить категорию.**

In [20]:
def categorize_purpose(row):
    try:
        if 'автом' in row:
            return 'операции с автомобилем'
        elif 'жил' in row or 'недвиж' in row:
            return 'операции с недвижимостью'
        elif 'свад' in row:
            return 'проведение свадьбы'
        elif 'образов' in row:
            return 'получение образования'
    except:
        return 'нет категории'

In [21]:
data['purpose_category'] = data['purpose'].apply(categorize_purpose)

### Шаг 3. Исследуйте данные и ответьте на вопросы

#### 3.1 Есть ли зависимость между количеством детей и возвратом кредита в срок?

In [22]:
people_with_debts = data.groupby('children')['debt'].sum()
people_with_debts.rename('people_with_debts')
people_with_debts

children
0    1063
1     444
2     194
3      27
4       4
5       0
Name: debt, dtype: int64

In [23]:
people_with_children = data.groupby('children')['debt'].count()
people_with_children

children
0    14091
1     4808
2     2052
3      330
4       41
5        9
Name: debt, dtype: int64

In [24]:
children_info = people_with_debts / people_with_children
children_info = children_info.sort_values(ascending=False)
children_info

children
4    0.097561
2    0.094542
1    0.092346
3    0.081818
0    0.075438
5    0.000000
Name: debt, dtype: float64

In [25]:
pivot_table = pd.concat([people_with_debts, people_with_children, children_info], axis=1)
pivot_table.columns = ['people_with_debts', 'people_with_children', 'children_info']
pivot_table

people_with_debts  people_with_children  children_info
children                                                        
0                      1063                 14091       0.075438
1                       444                  4808       0.092346
2                       194                  2052       0.094542
3                        27                   330       0.081818
4                         4                    41       0.097561
5                         0                     9       0.000000

<div class="alert-success" style="border-radius: 10px; padding: 10px">
<b>Комментарий ревьюера V2 ✔️</b>

Это правильная сводная таблица со всеми нужными для сравнения категорий данными. 👌</div>

<div class="alert-danger" style="padding: 20px; border-radius: 10px">
<b>Комментарий ревьюера ❌</b>

Ты правильно все посчитал, но одним из ключевых моментов этого проекта является умение создавать сводные таблицы. Подумай, как можно сразу вывести нужные категории, количество заемщиков в них, и процент просрочки в одной таблице.</div>

<div class="alert-info" style="padding: 20px; border-radius: 10px">
<b>Комментарий студента 💎</b>

Здесь исправил. Добавил сводную таблицу</div>

**Вывод:** 
При простой группировке может показаться, что, чем меньше детей, тем больше людей которые не вернули кредит в долг, но если учесть, что количество людей без детей например 14091, а людей с оним ребенком уже только 4808 и так далее, то можем найти процентное соотношение. Полученные значения говорят о том, что чаще всего долгжниками становятся люди с двумя и четыремя детьми, реже должниками становятся люди без детей. Людей с пятью детьми всего 9 человек, и делать вывод о том что все они точно не будут должниками не верно. Если же рассмотреть только три самые многочисленные выборки (люди у которых 1, 2 или нет детей) можем увидеть, чем больше детей у людей тем чаще они становятся должниками.

<div class="alert-warning" style="padding: 20px; border-radius: 10px">
<b>Комментарий ревьюера ⚠️</b>

Ты правильно обратил внимание на то, что выборки несбалансированны. Поэтому выявлять зависимость стоит, сравнивая самые большие, сопоставимые по размеру категории заемщиков. Какую зависимость можно выявить, сравнив категории с 0, 1 и 2 детьми?</div>

<div class="alert-info" style="padding: 20px; border-radius: 10px">
<b>Комментарий студента 💎</b>

Здесь исправил. Дополнил вывод, рассмотрев самые многочисленные выборки</div>

<div class="alert-success" style="border-radius: 10px; padding: 10px">
<b>Комментарий ревьюера V2 ✔️</b>

Да, все верно.</div>

#### 3.2 Есть ли зависимость между семейным положением и возвратом кредита в срок?

In [26]:
people_with_debts_family_stat = data.groupby('family_status')['debt'].sum()
people_with_debts_family_stat

family_status
Не женат / не замужем    273
в разводе                 84
вдовец / вдова            63
гражданский брак         385
женат / замужем          927
Name: debt, dtype: int64

In [27]:
people_with_family_stat = data.groupby('family_status')['debt'].count()
people_with_family_stat

family_status
Не женат / не замужем     2796
в разводе                 1189
вдовец / вдова             951
гражданский брак          4134
женат / замужем          12261
Name: debt, dtype: int64

In [28]:
status_inf = people_with_debts_family_stat / people_with_family_stat
status_inf.sort_values(ascending=False)

family_status
Не женат / не замужем    0.097639
гражданский брак         0.093130
женат / замужем          0.075606
в разводе                0.070648
вдовец / вдова           0.066246
Name: debt, dtype: float64

In [29]:
pivot_table = pd.concat([people_with_debts_family_stat, people_with_family_stat, status_inf], axis=1)
pivot_table.columns = ['people_with_debts_family_stat', 'people_with_family_stat', 'status_info']
pivot_table

people_with_debts_family_stat  people_with_family_stat  \
family_status                                                                   
Не женат / не замужем                            273                     2796   
в разводе                                         84                     1189   
вдовец / вдова                                    63                      951   
гражданский брак                                 385                     4134   
женат / замужем                                  927                    12261   

                       status_info  
family_status                       
Не женат / не замужем     0.097639  
в разводе                 0.070648  
вдовец / вдова            0.066246  
гражданский брак          0.093130  
женат / замужем           0.075606

<div class="alert-success" style="border-radius: 10px; padding: 10px">
<b>Комментарий ревьюера V2 ✔️</b>

Отлично. 👌</div>

<div class="alert-warning" style="padding: 20px; border-radius: 10px">
<b>Комментарий ревьюера ⚠️</b>

Собери пожалуйста все данные в одну сводную таблицу.</div>

<div class="alert-info" style="padding: 20px; border-radius: 10px">
<b>Комментарий студента 💎</b>

Здесь исправил.</div>

**Вывод:** Повторим наши рассуждения для поиска корреляции между семейным положением и возвратом кредита в срок. Из полученных результатов видно, что не женатые и не замужние люди, а так же люди в гражданском браке чаще не возвращают кредит в срок, а вдовцы и вдовы реже. Но разброс полученных значений не велик, не женатых и не замужних людей не вернувших долги в срок 9.7%, а вдовцов и вдов 6.6%

<div class="alert-success" style="padding: 20px; border-radius: 10px">
<b>Комментарий ревьюера ✔️</b>

Верно, можно сказать, что состоящие или состоявшие в браке заемщики реже допускают просрочки.</div>

#### 3.3 Есть ли зависимость между уровнем дохода и возвратом кредита в срок?

**Для удобства приведем еще раз категории людей по уровню дохода:**
- 0–30000 — `'E'`;
- 30001–50000 — `'D'`;
- 50001–200000 — `'C'`;
- 200001–1000000 — `'B'`;
- 1000001 и выше — `'A'`.

In [30]:
people_with_debts_income = data.groupby('total_income_category')['debt'].sum()
people_with_debts_income

total_income_category
A       2
B     354
C    1353
D      21
E       2
Name: debt, dtype: int64

In [31]:
people_with_income_cat = data.groupby('total_income_category')['debt'].count()
people_with_income_cat

total_income_category
A       25
B     5014
C    15921
D      349
E       22
Name: debt, dtype: int64

In [32]:
income_inf = people_with_debts_income / people_with_income_cat
income_inf.sort_values(ascending=False)

total_income_category
E    0.090909
C    0.084982
A    0.080000
B    0.070602
D    0.060172
Name: debt, dtype: float64

In [33]:
pivot_table = pd.concat([people_with_debts_income, people_with_income_cat, income_inf], axis=1)
pivot_table.columns = ['people_with_debts_income', 'people_with_income_cat', 'income_inf']
pivot_table

people_with_debts_income  people_with_income_cat  \
total_income_category                                                     
A                                             2                      25   
B                                           354                    5014   
C                                          1353                   15921   
D                                            21                     349   
E                                             2                      22   

                       income_inf  
total_income_category              
A                        0.080000  
B                        0.070602  
C                        0.084982  
D                        0.060172  
E                        0.090909

**Вывод:** Повторим наши рассуждения для поиска корреляции между уровнем дохода и возвратом кредита в срок. Из полученных результатов видно, что люди с доходом ниже 30000 чаще не возвращают кредит в срок, а люди с доходом от 30001 до 50000 реже становятся должниками. Если же сравнить две самые многочисленные выборки, а это люди с доходом от 200001 до 1000000 (категория B) и с доходом от 50001 до 200000 (категория С), то можно заметить что люди с большим доходом чаще становятся должниками.

<div class="alert-warning" style="padding: 20px; border-radius: 10px">
<b>Комментарий ревьюера ⚠️</b>

А какую можно выявить зависимость, если сравнить только две самые многочисленные выборки?

+сводная таблица</div>

<div class="alert-info" style="padding: 20px; border-radius: 10px">
<b>Комментарий студента 💎</b>

Здесь исправил.</div>

<div class="alert-success" style="border-radius: 10px; padding: 10px">
<b>Комментарий ревьюера V2 ✔️</b>

Верно, если сравнить самые большие категории B и C, можно сказать, что заемщики с более высоким доходом платят кредиты вовремя лучше.

Сводная таблица - правильная.</div>

#### 3.4 Как разные цели кредита влияют на его возврат в срок?

In [34]:
people_with_debts_purpose = data.groupby('purpose_category')['debt'].sum()
people_with_debts_purpose

purpose_category
операции с автомобилем      400
операции с недвижимостью    780
получение образования       369
проведение свадьбы          183
Name: debt, dtype: int64

In [35]:
people_with_purpose = data.groupby('purpose_category')['debt'].count()
people_with_purpose

purpose_category
операции с автомобилем       4279
операции с недвижимостью    10751
получение образования        3988
проведение свадьбы           2313
Name: debt, dtype: int64

In [36]:
purpose_inf = people_with_debts_purpose / people_with_purpose
purpose_inf.sort_values(ascending=False)

purpose_category
операции с автомобилем      0.093480
получение образования       0.092528
проведение свадьбы          0.079118
операции с недвижимостью    0.072551
Name: debt, dtype: float64

In [37]:
pivot_table = pd.concat([people_with_debts_purpose, people_with_purpose, purpose_inf], axis=1)
pivot_table.columns = ['people_with_debts_purpose', 'people_with_purpose', 'purpose_inf']
pivot_table

people_with_debts_purpose  people_with_purpose  \
purpose_category                                                           
операции с автомобилем                          400                 4279   
операции с недвижимостью                        780                10751   
получение образования                           369                 3988   
проведение свадьбы                              183                 2313   

                          purpose_inf  
purpose_category                       
операции с автомобилем       0.093480  
операции с недвижимостью     0.072551  
получение образования        0.092528  
проведение свадьбы           0.079118

<div class="alert-success" style="border-radius: 10px; padding: 10px">
<b>Комментарий ревьюера V2 ✔️</b>

Супер!</div>

<div class="alert-warning" style="padding: 20px; border-radius: 10px">
<b>Комментарий ревьюера ⚠️</b>

Здесь тоже нужно собрать все данные в одну сводную таблицу.</div>

<div class="alert-info" style="padding: 20px; border-radius: 10px">
<b>Комментарий студента 💎</b>

Здесь исправил.</div>

**Вывод:** Повторим наши рассуждения для поиска корреляции между целью кредита и возвратом кредита в срок. Из полученных результатов видно, что люди, которые  берут кредит для операций с автомобилем или для получения образования чаще не возвращают кредит в срок, а люди, которые берут кредит для проведения свадьбы или операций с недвижимостью реже становятся должниками.

<div class="alert-success" style="padding: 20px; border-radius: 10px">
<b>Комментарий ревьюера ✔️</b>

Правильно, самыми проблемными выглядят автокредиты и кредиты на образование. </div>

#### 3.5 Приведите возможные причины появления пропусков в исходных данных.

*Ответ:* Самая вероятная причина появления пропусков это ошибки сотрудников, не заметивших пропущенные поля. Но так же возможна потеря данных при переносе таблиц, или в результате сбоя в работе компьютера.

<div class="alert-warning" style="padding: 20px; border-radius: 10px">
<b>Комментарий ревьюера ⚠️</b>

Ты правильно указал человеческий фактор, как наиболее вероятную причину образования пропусков в данных. Но какая причина еще может вызвать потерю данных?</div>

<div class="alert-info" style="padding: 20px; border-radius: 10px">
<b>Комментарий студента 💎</b>

Здесь исправил.</div>

<div class="alert-success" style="border-radius: 10px; padding: 10px">
<b>Комментарий ревьюера V2 ✔️</b>

Согласен, техническую ошибку также исключать нельзя.</div>

#### 3.6 Объясните, почему заполнить пропуски медианным значением — лучшее решение для количественных переменных.

*Ответ:* Так как если в данных присутствуют различного рода выбросы (слишком большое значени или же наоборот, слишком маленькое), то если заменять пропуски средним это может исказить реадьные данные. Именно поэтому используется медианное значение.

<div class="alert-success" style="padding: 20px; border-radius: 10px">
<b>Комментарий ревьюера ✔️</b>

Совершенно верно, медиана менее чувствительна к выбросам.</div>

### Шаг 4: общий вывод.

**Была поставлена задача разобраться, влияет ли семейное положение и количество детей клиента на факт погашения кредита в срок. Входные данные от банка — статистика о платёжеспособности клиентов.**
**В таблице были представленны следующие данные:**

- `children` — количество детей в семье
- `days_employed` — общий трудовой стаж в днях
- `dob_years` — возраст клиента в годах
- `education` — уровень образования клиента
- `education_id` — идентификатор уровня образования
- `family_status` — семейное положение
- `family_status_id` — идентификатор семейного положения
- `gender` — пол клиента
- `income_type` — тип занятости
- `debt` — имел ли задолженность по возврату кредитов
- `total_income` — ежемесячный доход
- `purpose` — цель получения кредита
    
**Далее была проведена обработка полученных данных. Были заполненны пропуски в столбцах `total_income` и `days_employed` с помощью медианных значений по соответсвующим `income_type`. Так же были удалены дубликаты.** 
**Из предоставленной статистики можно сделать следующие выводы:**

- Из полученных результатов видно, что не женатые и не замужние люди, а так же люди в гражданском браке чаще не возвращают кредит в срок, а вдовцы и вдовы реже;
- Чаще всего долгжниками становятся люди с двумя и четыремя детьми, реже должниками становятся люди без детей.
- Цели, для которых люди берут кредит, так же влияют на возврат денег во время, но разброс количества людей не большой от 7.2% до 9.3%

**Я бы порекомендовал банку более внимательно относится к клиентам, которые удовлетворяют следующим критериям:**

- Берут кредит для операций с недвижимостью или для проведения свадьбы
- Имеют одного или двух детей
- Имеют доход от 50001 до 200000

**Так как именно эти категории людей чаще всего становятся должниками**


<div class="alert-success" style="border-radius: 10px; padding: 10px">
<b>Комментарий ревьюера V2 ✔️</b>

Теперь у тебя получился подробный и правильный вывод, отражающий все основные результаты исследования. Молодец, что не забыл про рекомендации. 👍</div>

<div class="alert-warning" style="padding: 20px; border-radius: 10px">
<b>Комментарий ревьюера ⚠️</b>

Общий вывод имеет смысл немного расширить. Нужно отразить в нем основные результаты по всем шагам работы. Начать с абзаца с описанием входных данных и что сделали на этапе предобработки. 

Далее описать все выявленные зависимости просрочки от различных факторов в общем виде. 
    
Отдельные формулировки надо будет поправить после доработки проекта.

Закончить вывод можно рекомендациями банку по улучшению сбора данных.

Зачастую, заказчик читает только общий вывод, поэтому он должен быть максимально информативным.

</div>

<div class="alert-info" style="padding: 20px; border-radius: 10px">
<b>Комментарий студента 💎</b>

Здесь исправил.</div>

<div style="border:solid steelblue 3px; padding: 20px; border-radius: 10px">
    
<b>Общий комментарий ревьюера:</b>

Ты проделал большую работу, но нужно кое-что исправить, чтобы я ее принял.
    
Основное критическое замечание - отсутствуют сводные таблицы с нужными для сравнения категорй данными.
    
Также, постарайся отработать желтые комментарии во второй части работы.
    
Буду ждать твой доработанный проект. ⌛️</div>

<div style="border:solid steelblue 3px; padding: 20px; border-radius: 10px">
    
<b>Общий комментарий ревьюера V2</b>

Спасибо, что доработал проект! Критических замечаний больше нет, получилась отличная работа.
    
Успехов в дальнейшем обучении! 🏆
    
</div>